sql/sql_alter_instance.cc

```diff
@@ -186,7 +186,15 @@ bool Rotate_innodb_master_key::execute() {
 */
 extern bool wsrep_rotate_master_key();
 bool Rotate_gcache_master_key::execute() {
-  if (check_security_context()) return true;
+
+  Security_context *sctx = m_thd->security_context();
+  if (!sctx->check_access(SUPER_ACL) &&
+      !sctx->has_global_grant(STRING_WITH_LEN("BINLOG_ENCRYPTION_ADMIN"))
```

Comment:
Why `BINLOG_ENCRYPTION_ADMIN`? @kamil-holubicki Can you please confirm if we need `BINLOG_ENCRYPTION_ADMIN` privilege to enable gcache encryption.

Reply:
It has to be the same as in `Rotate_innodb_master_key::execute()`, so we need `ENCRYPTION_KEY_ADMIN`